### PR TITLE
[v1.x] Onnx fix slice_axis and embedding and reshape

### PR DESCRIPTION
**Pull request by Zha0q1**
### **fixed:**

1. slice_axis
2. reshape
3. embedding

### **added/tweaked tests for** 
1. embedding
2. slice_axis
3. reshape
4. contrib_interleaved_matmul_selfatt_qk

### CODE REVIEW

File: python/mxnet/contrib/onnx/mx2onnx/_op_translations.py [hunk 1]
```diff
@@ -1645,30 +1645,37 @@ def convert_cast(node, **kwargs):
...
+    create_tensor([begin], name+'_begin',kwargs["initializer"])
+    if not end or end == 'None':
         # ONNX doesn't support None for ends. Since ends=None depicts
         # length of dimension, passing dimension in this case.
-        in_shape = kwargs['in_shape'][0]
-        ends = in_shape[axes]
+        create_tensor([axis+1], name+"_axis_plus_1", kwargs["initializer"])
+        nodes += [
+            make_node('Shape', [input_nodes[0]], [name+"_data_shape"]),
+            make_node('Slice', [name+'_data_shape', name+'_axis', name+'_axis_plus_1'],
```
Comment (waytrue17) [inline python/mxnet/contrib/onnx/mx2onnx/_op_translations.py:1667]: when `end` is not specified, would the default value be the length of that dimension?

Comment (Zha0q1) [reply python/mxnet/contrib/onnx/mx2onnx/_op_translations.py:1667]: yes. this is covered by a test case too

**Review by josephevans — APPROVED** [review]
LGTM!

**Review by waytrue17 — APPROVED** [review]
LGTM

**Review by josephevans — APPROVED** [review]

**Review by waytrue17 — APPROVED** [review]
LGTM!

**Review by josephevans — APPROVED** [review]
LGTM!

**Pull request merged**